### PR TITLE
Make static-context and lazy-static-context addictive

### LINKED ISSUE #77: Poor usage of Cargo features for `static-context` vs `lazy-static-context`
It is currently forbidden to enable both the `static-context` and `lazy-static-context` features at the same time: https://docs.rs/libsecp256k1/0.6.0/src/libsecp256k1/lib.rs.html#41

This is strongly discouraged, as documented [here](https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features).

### PR DESCRIPTION
**Pull request by sorpaas**
closes #77

### CODE REVIEW

**Comment by athei** [comment]
I hate to be this guy but we need a `README` update regarding this feature 😅

File: src/lib.rs [hunk 1]
```diff
@@ -49,13 +46,13 @@ lazy_static::lazy_static! {
     pub static ref ECMULT_GEN_CONTEXT: Box<ECMultGenContext> = ECMultGenContext::new_boxed();
 }
 
-#[cfg(feature = "static-context")]
+#[cfg(all(feature = "static-context", not(feature = "lazy-static-context")))]
 /// A static ECMult context.
 // Correct `pre_g` values are fed into `ECMultContext::new_from_raw`, generated by build script.
```
Comment (dvdplm) [inline src/lib.rs:51]:
```suggestion
/// A static ECMult context. Requires that the `static-context` to be enabled and that `lazy-static-context` is **not** enabled.
/// Correct `pre_g` values are fed into `ECMultContext::new_from_raw`, generated by build script.
```

File: src/lib.rs [hunk 2]
```diff
@@ -49,13 +46,13 @@ lazy_static::lazy_static! {
...
 pub static ECMULT_CONTEXT: ECMultContext =
     unsafe { ECMultContext::new_from_raw(include!(concat!(env!("OUT_DIR"), "/const.rs"))) };
 
-#[cfg(feature = "static-context")]
+#[cfg(all(feature = "static-context", not(feature = "lazy-static-context")))]
 /// A static ECMultGen context.
```
Comment (dvdplm) [inline src/lib.rs:56]: Same note in the docs here.

**Review by dvdplm — APPROVED** [review]
Definitely would be good to document the features offered in the readme.

File: src/lib.rs (hunk 1 above)
Comment (sorpaas) [reply src/lib.rs:51]: `ECMULT_CONTEXT` is defined either here or through `lazy_static`. It is always available either with `static-context` or `lazy-static-context`.

The second line is related to implementation details of how this is generated, so it doesn't really fit the docs.

File: src/lib.rs (hunk 2 above)
Comment (sorpaas) [reply src/lib.rs:56]: Same as above.

**Review by athei — APPROVED** [review]

**Pull request merged**